[ACI] Updating summary/description fields to generate documentation

jluk opened the pull request:
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process. 

### PR information
- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [x] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [x] If applicable, the PR references the bug/issue that it fixes.
- [x] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [x] My spec meets the review criteria:
  - [x] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [x] The spec follows the guidelines described in the [Swagger checklist](../documentation/swagger-checklist.md) document.
  - [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

Commit:
first pass on improving descriptions for rest api doc gen

Commit:
Merge branch 'current' of https://github.com/Azure/azure-rest-api-specs into current

msftclas commented:
@jluk,
Thanks for your contribution as a Microsoft full-time employee or intern. You do not need to sign a CLA.
_Thanks,
Microsoft Pull Request Bot_

jluk commented on specification/containerinstance/resource-manager/Microsoft.ContainerInstance/2017-08-01-preview/containerInstance.json:
```diff
@@ -606,12 +612,12 @@
                 },
                 "osType": {
                   "type": "string",
-                  "description": "The base level OS type required by the containers in the group.",
```
What do you mean by "base level" - host-level? Does the user need to know that, can they just input the OS type of the image they bring?

jluk commented on specification/containerinstance/resource-manager/Microsoft.ContainerInstance/2017-08-01-preview/containerInstance.json:
```diff
@@ -596,7 +602,7 @@
                 },
                 "restartPolicy": {
                   "type": "string",
-                  "description": "- `always` Always restart\n",
```
Currently always is the only option and defaulted, desc should update when r2c/never are implemented.

jluk commented on specification/containerinstance/resource-manager/Microsoft.ContainerInstance/2017-08-01-preview/containerInstance.json:
```diff
@@ -482,79 +488,79 @@
       ],
       "properties": {
         "shareName": {
-          "description": "The Azure file share name.",
```
We should decide on whether we want to call it "volume" or "share" AFAIK its interchangeable?

jluk commented on specification/containerinstance/resource-manager/Microsoft.ContainerInstance/2017-08-01-preview/containerInstance.json:
```diff
@@ -302,68 +308,68 @@
         },
         "instanceView": {
-          "description": "The instance view of the container. Only valid in response.",
+          "description": "The instance view of the container instance. Only valid in response.",
```
Throughout all references to a single container, I updated it to state "container instance".

jluk commented on specification/containerinstance/resource-manager/Microsoft.ContainerInstance/2017-08-01-preview/containerInstance.json:
```diff
@@ -678,7 +684,7 @@
+          "description": "The type of IP address determining whether the IP is exposed to the public internet.",
           "enum": [
             "Public"
           ],
```
On Line 709 - I don't think you can actually specify a UDP port today. We should remove this bit if we generate docs before it is supported.

jluk replied:
Do we know when UDP will be supported, we can leave the documentation for it if it is very soon.

jluk commented on specification/containerinstance/resource-manager/Microsoft.ContainerInstance/2017-08-01-preview/containerInstance.json:
```diff
@@ -41,7 +41,8 @@
           }
         },
-        "description": "Get the list of container groups in a given subscription.",
+        "summary": "Get a list of container groups in the specified subscription.",
```
Added summary fields on each explicit rest call to conform to how ACS gens docs.

jluk commented on specification/containerinstance/resource-manager/Microsoft.ContainerInstance/2017-08-01-preview/containerInstance.json:
```diff
@@ -41,7 +41,8 @@
         },
-        "description": "Get the list of container groups in a given subscription.",
+        "summary": "Get a list of container groups in the specified subscription.",
+        "description": "Get a list of container groups in the specified subscription. This operation returns properties of each container group including containers, image registry credentials, restart policy, IP address type, OS type, state, and volumes.",
```
Added each property of the container group to conform to how ACS describes each service.

jluk commented on specification/containerinstance/resource-manager/Microsoft.ContainerInstance/2017-08-01-preview/containerInstance.json:
```diff
@@ -52,7 +53,7 @@
         "responses": {
           "200": {
-            "description": "Successful response",
+            "description": "OK",
```
Update 200 response descriptions to OK for consistency

jluk commented on specification/containerinstance/resource-manager/Microsoft.ContainerInstance/2017-08-01-preview/containerInstance.json:
```diff
@@ -201,7 +205,8 @@
         },
-        "description": "Delete container groups.",
+        "summary": "Delete the specified container groups.",
+        "description": "Delete the specified container groups in the specified subscription and resource group. The operation does not delete other resources created as part of creating a container group such as volumes. All other resources created with the container group are part of the same resource group and can be deleted individually.",
```
Is this added detail correct about not deleting the volumes associated and any IP addresses/etc. being in the same RG?

veronicagg replied:
Should be fine, as long as it's helpful for the customer.

yolo3301 replied:
Volumes/image registry credentials are provided by the user when creating a container group. But IP is different. It's provided by ACI. Once a container group is deleted, the IP will be recycled. I think it would be more clear to say `The operation does not delete any resources provided by the users such as volumes`

jluk commented:
@yolo3301 could you review this PR/comments when you get a free moment? Once we improve on the descriptions/summaries a bit we can generate the docs for more user feedback.

jluk reviewed (commented):
Leaving comments here for @yolo3301 to review description/summary updates.

veronicagg reviewed (changes requested):
Changes look good to me. 
Per request from @jluk, PR is pending review from @yolo3301 . 
Once I get confirmation from you that the PR look good, I can approve and merge. Thanks!

veronicagg commented on specification/containerinstance/resource-manager/Microsoft.ContainerInstance/2017-08-01-preview/containerInstance.json:
```diff
@@ -245,7 +250,8 @@
           }
         },
-        "description": "Get the logs for this container.",
+        "summary": "Get the logs for a specified container instance in a specified resource group and container group.",
```
You may want to address this warning: https://travis-ci.org/Azure/azure-rest-api-specs/jobs/270502075#L673
"The summary and description values should not be same."

mmacy commented on specification/containerinstance/resource-manager/Microsoft.ContainerInstance/2017-08-01-preview/containerInstance.json:
```diff
@@ -482,79 +488,79 @@
     },
     "AzureFileVolume": {
-      "description": "The Azure file volume.",
+      "description": "The properties of the Azure File volume.",
```
Might be good to mention that an Azure file *share* is mounted as a volume.

mmacy commented on specification/containerinstance/resource-manager/Microsoft.ContainerInstance/2017-08-01-preview/containerInstance.json:
```diff
@@ -482,79 +488,79 @@
         },
         "storageAccountKey": {
-          "description": "The storage account key to access to the Azure file.",
+          "description": "The storage account key used to access the Azure File volume.",
```
storage account *access* key

mmacy commented on specification/containerinstance/resource-manager/Microsoft.ContainerInstance/2017-08-01-preview/containerInstance.json:
```diff
@@ -482,79 +488,79 @@
         },
         "storageAccountName": {
-          "description": "The storage account name to access to the Azure file.",
+          "description": "The name of the storage account that contains the Azure File volume.",
```
Another volume vs. share comment, Azure Files does not have volumes, but shares. We definitely want to be careful not to mix terms. I'll refrain from further volume/share comments, you get the idea. :)

mmacy commented on specification/containerinstance/resource-manager/Microsoft.ContainerInstance/2017-08-01-preview/containerInstance.json:
```diff
@@ -482,79 +488,79 @@
         },
         "mountPath": {
-          "description": "The path with in the container at which the volume should be mounted. Must not contain ':'.",
+          "description": "The path within the container where the volume should be mounted. Must not contain ':' (colon).",
```
Must not contain colon (:).

(list full character name first, then the character in parens, no quotes)

mmacy commented on specification/containerinstance/resource-manager/Microsoft.ContainerInstance/2017-08-01-preview/containerInstance.json:
```diff
@@ -678,7 +684,7 @@
         "type": {
           "type": "string",
-          "description": "Determines whether the IP is exposed to the public internet.",
+          "description": "The type of IP address determining whether the IP is exposed to the public internet.",
```
Recommend simply striking "The type of IP address determining" from this, or go with something like:

"Specifies whether the IP is exposed to the public internet."

mmacy replied:
Also, line **678** (the "ports" property description) could do with a bit more explanatory text. Currently, it reads:

"The ports."

TCP only? UDP? Both? How do ports specified here relate to those on a container instance? Doesn't have to be long and involved, but a little more on this could go a long way.

mmacy reviewed (commented):
This is looking way better! Just a few comments.

yolo3301 commented on specification/containerinstance/resource-manager/Microsoft.ContainerInstance/2017-08-01-preview/containerInstance.json:
```diff
@@ -385,35 +391,35 @@
         "exitCode": {
           "type": "integer",
-          "description": "The container exit code."
+          "description": "The container instance exit code."
```
It will follow the `docker run` [exit code](https://docs.docker.com/engine/reference/run/#exit-status). Maybe reference that?

yolo3301 commented:
@jluk I left a couple of comments. Mostly look good to me.

jluk committed:
update on feedback 1

address comments, need clarity on ports vs containerPort descriptions

jluk commented:
@yolo3301 made updates lmk thoughts. qq - what is the difference between port and containerPort?

jluk commented on specification/containerinstance/resource-manager/Microsoft.ContainerInstance/2017-08-01-preview/containerInstance.json:
```diff
@@ -704,7 +704,7 @@
     },
     "Port": {
-      "description": "The port.",
+      "description": "The port exposed on the container group.",
```
Is this accurate?

jluk replied:
@yolo3301 from what I am reading in the swagger, "ports" are the ports utilized in the container group while "containerPorts" refer to individual instances. Could you verify my references are correct?

yolo3301 replied:
You're right. Ports under container group are the ports exposed with the public IP. Instance ports can be exposed for container-wise communication.

mmacy commented on specification/containerinstance/resource-manager/Microsoft.ContainerInstance/2017-08-01-preview/containerInstance.json:
```diff
@@ -201,7 +205,8 @@
         },
-        "description": "Delete container groups.",
+        "summary": "Delete the specified container groups.",
+        "description": "Delete the specified container groups in the specified subscription and resource group. The operation does not delete other resources provided by the user such as volumes.",
```
(very) Minor: "by the user, such as volumes." (add comma)

mmacy commented on specification/containerinstance/resource-manager/Microsoft.ContainerInstance/2017-08-01-preview/containerInstance.json:
```diff
@@ -385,35 +391,35 @@
         "exitCode": {
           "type": "integer",
-          "description": "The container exit code."
+          "description": "The container instance exit code. Exit codes map to output from `docker run`."
```
How about:

"Exit codes correspond to those of the `docker run` command."

It feels a bit odd that that we're referencing an external tool in our own API reference documentation, but in this case, if it provides the best and most succinct explanation, it might be appropriate. Not sure there's another way to better communicate this, but wanted to call it out as a potential gotcha we might hit later on.

jluk replied:
@yolo3301 is there another way to reference the exit codes that can abstract us from docker?

mmacy commented on specification/containerinstance/resource-manager/Microsoft.ContainerInstance/2017-08-01-preview/containerInstance.json:
```diff
@@ -596,7 +602,7 @@
                 "restartPolicy": {
                   "type": "string",
-                  "description": "- `always` Always restart\n",
+                  "description": "Restart policy for all containers within the container group. Only option is Always.",
```
The enum value is "always," with lowercase 'a' -- should we maintain consistency with that? Also, a slightly modified option here, but might make it sound less like a limit, and perhaps even hint at future options:

"Restart policy for all containers within the container group. The currently available option is `always`."

mmacy reviewed (approved):
Looking great! Just a couple suggestions, but no blockers.

One thing I didn't comment on was ContainerPorts and possibly adding a bit of differentiation text between that and the container group port definition (can save this edit for a future quality pass).

jluk committed:
Include feedback from Marsh

Added comma, update docker run ref, update enum always, update containerPort/port descriptions

yolo3301 commented on specification/containerinstance/resource-manager/Microsoft.ContainerInstance/2017-08-01-preview/containerInstance.json:
```diff
@@ -256,34 +262,34 @@
             "name": "tail",
             "in": "query",
-            "description": "Only show this number of log lines. If not provided, all available logs are shown.",
+            "description": "The number of lines to show from the container instance log. If not provided, all available logs are shown.",
```
Maybe explicitly say about `tail`? Also if not provided, I think it will show up to 4mb logs. No more than that.

jluk replied:
Good feedback - added a bit about logs coming from the tail and the 4mb limit

jluk committed:
detail tail updates

wenwu449 commented on specification/containerinstance/resource-manager/Microsoft.ContainerInstance/2017-08-01-preview/containerInstance.json:
```diff
@@ -158,20 +162,20 @@
           {
             "name": "containerGroup",
-            "description": "Definition of the container to be created.",
+            "description": "The properties of the container group to be created.",
```
"to be created or updated"

wenwu449 commented on specification/containerinstance/resource-manager/Microsoft.ContainerInstance/2017-08-01-preview/containerInstance.json:
```diff
@@ -181,13 +185,13 @@
         "responses": {
           "200": {
-            "description": "OK - the container group was previously created and is now updated",
+            "description": "OK - the container group already exists and is now updated.",
```
if there is no spec change, we also return 200, so 200 may be a no-op. Not sure if this information need to be exposed to customer.

jluk replied:
Good point - removing additional messages from 200 responses to be only "OK".

wenwu449 commented on specification/containerinstance/resource-manager/Microsoft.ContainerInstance/2017-08-01-preview/containerInstance.json:
```diff
@@ -201,7 +205,8 @@
           }
         },
-        "description": "Delete container groups.",
+        "summary": "Delete the specified container groups.",
```
we only support delete on single container group, do we need the plural form groups?

jluk replied:
Agree, updating the description. However why are all the operationIds plural (ContainerGroups_Delete)? I suppose the plural form is the default ARM resource name is why?

jluk committed:
pass 3 with Wenjun's edits

jluk commented:
@yolo3301 please make a pass and sign-off for @veronicagg to merge if it looks good. thanks everyone for the quick edits!

yolo3301 commented:
@veronicagg I have approved the PR. Please help merge. Thanks!

AutorestCI commented:
No modification for AutorestCI/azure-sdk-for-ruby

AutorestCI commented:
No modification for AutorestCI/azure-sdk-for-node

AutorestCI commented:
[![Build Status](https://travis-ci.org/AutorestCI/azure-sdk-for-python.svg?branch=RestAPI-PR1621)](https://travis-ci.org/AutorestCI/azure-sdk-for-python) https://github.com/Azure/azure-sdk-for-python/pull/1420
